pkg/proxy/apis/config/validation/validation.go

```diff
@@ -95,6 +95,8 @@ func Validate(config *kubeproxyconfig.KubeProxyConfiguration) field.ErrorList {
 
 	allErrs = append(allErrs, validateKubeProxyNodePortAddress(config.NodePortAddresses, newPath.Child("NodePortAddresses"))...)
 	allErrs = append(allErrs, validateShowHiddenMetricsVersion(config.ShowHiddenMetricsForVersion, newPath.Child("ShowHiddenMetricsForVersion"))...)
+
+	allErrs = append(allErrs, validateDetectLocalMode(config.DetectLocalMode, newPath.Child("DetectLocalMode"))...)
```

Comment:
IIUC this moves the list of which local-detectors are supported from the non-Windows code into all-platforms space.  Is that what we want?  Does Windows support LocalModeBridgeInterface, for example?

Reply:
`--detect-local-mode` is currently a Linux-only option. Windows users _should_ just not pass that option at all (which `validateDetectLocalMode` will consider valid).

The behavioral change here is that before, on Windows, you could do `--detect-local-mode asdfasdfasdfasdfas` and kube-proxy would accept it, whereas now if you do that, it will error out telling you that that isn't a valid mode.

In theory we could validate that Windows users aren't passing Linux-specific options, but in practice, winkernel currently uses `--iptables-sync-period` and `--iptables-min-sync-period` to configure its BoundedFrequencyRunner. (Sigh. I'll have a fix for that at some point...)

Reply:
(Probably we should have made `--detect-local-mode` be two separate options, `--iptables-detect-local-mode` and `--ipvs-detect-local-mode`...)

Reply:
😱  and `--nftables-detect-local-mode`  :)

Reply:
the tendency to deal with cross platform in kubernetes is to use `if runtime.GOOS == "windows"`

```
pkg/kubelet/kuberuntime/kuberuntime_sandbox.go: if runtime.GOOS == "windows" {
pkg/kubelet/kuberuntime/kuberuntime_sandbox.go:         if sc.RunAsUser != nil && runtime.GOOS != "windows" {
pkg/kubelet/kuberuntime/kuberuntime_sandbox.go:         if sc.RunAsGroup != nil && runtime.GOOS != "windows" {
pkg/kubelet/kuberuntime/kuberuntime_sandbox.go:         if sc.FSGroup != nil && runtime.GOOS != "windows" {
pkg/kubelet/kuberuntime/kuberuntime_sandbox.go:         if sc.SELinuxOptions != nil && runtime.GOOS != "windows" {
pkg/kubelet/sysctl/safe_sysctls.go:     if goruntime.GOOS == "linux" {
pkg/kubelet/kubelet_node_status.go:                             v1.LabelOSStable:      goruntime.GOOS,
pkg/kubelet/kubelet_node_status.go:                             kubeletapis.LabelOS:   goruntime.GOOS,
pkg/kubelet/kubelet_node_status.go:     if !osLabelExists || osName != goruntime.GOOS {
pkg/kubelet/kubelet_node_status.go:             node.Labels[v1.LabelOSStable] = goruntime.GOOS
pkg/kubelet/kubelet_server_journal.go:  if n.Boot != nil && runtime.GOOS == "windows" {
```

so you can have granularity here

Reply:
yeah, `--mode` validation is already platform-specific